### DISCUSSION TITLE
The state of vehicle reaches a Position and stops, and how to trigger the vehicle to complete the next Position

### DISCUSSION THREAD

**Discussion by jrt324** [openTCS/opentcs#99] [Q&A]
When a vehicle receives a transport order (assuming that there are two positions where it needs to stop: **Loading** and **Unloading**), when the vehicle arrives at the first position (**Loading**) and stops, what is the status of the vehicle and the DriveOrder at this point?

What is the state of these objects:

- **DriveOrder.State** 
- **DestinationState.State** 
- **Vehicle.State**

 If **Loading** Action is Completed(assum the vehicle has a Button can be trigger) , What command does OpenTCS need to receive, or what does it need to know about the completion of the "Loading" Action at this position before it dispatchs the vehicle to the next position?

**Comment by swltr** [comment]
* Once the vehicle (driver) reports that the vehicle has completed its movement command, including the loading operation, the drive order's state will be set to `FINISHED`.
* There is not `DestinationState`. You probably mean something different here...?
* The vehicle's state is whatever the vehicle driver reports it to be.

openTCS expects the same as for every other movement command: The vehicle driver needs to reports it as completed via `getProcessModel().commandExecuted()`. Once the movement command has been reported as completed, openTCS may send the next one.
